admin/src/main/java/edu/unc/lib/dl/admin/controller/ExportXMLController.java

```diff
@@ -0,0 +1,300 @@
+
+			List<BriefObjectMetadata> objects = resultResponse.getResultList();
+			objects.add(0, container);
+
+			for (BriefObjectMetadata object : objects) {
+				pids.add(object.getPid().getPid());
+			}
+		}
+
+		XMLExportRunnable runnable = new XMLExportRunnable(new XMLExportRequest(pids, exportRequest.getEmail()),
+				GroupsThreadStore.getUsername(), GroupsThreadStore.getGroups());
+
+		Thread thread = new Thread(runnable);
+		thread.start();
+
+		Map <String, String> response = new HashMap<>();
+		response.put("message", "Metadata export for " + pids.size()
+				+ " objects has begun, you will receive the data via email soon");
+		return response;
+	}
+
+	/**
+	 * Generates an XML document containing metadata for all objects in the provided list of PIDs.
+	 * 
+	 * @param exportRequest
+	 * @return
+	 * @throws IOException
+	 * @throws FedoraException
+	 */
+	@RequestMapping(value = "exportXML", method = RequestMethod.POST)
+	public @ResponseBody
+	Object exportSet(@RequestBody XMLExportRequest exportRequest) throws IOException, FedoraException {
+
+		XMLExportRunnable runnable = new XMLExportRunnable(
+				exportRequest, GroupsThreadStore.getUsername(), GroupsThreadStore.getGroups());
+
+		Thread thread = new Thread(runnable);
+		thread.start();
+
+		Map <String, String> response = new HashMap<>();
+		response.put("message", "Metadata export for " + exportRequest.getPids().size()
+				+ " has begun, you will receive the data via email soon");
+		return response;
+	}
+
+	public static class XMLExportRequest {
+		private List<String> pids;
+		private String email;
+
+		public XMLExportRequest() {
+		}
+
+		public XMLExportRequest(List<String> pids, String email) {
+			this.pids = pids;
+			this.email = email;
+		}
+
+		public List<String> getPids() {
+			return pids;
+		}
+
+		public void setPids(List<String> pids) {
+			this.pids = pids;
+		}
+
+		public String getEmail() {
+			return email;
+		}
+
+		public void setEmail(String email) {
+			this.email = email;
+		}
+	}
+
+	/**
+	 * Runnable which performs the work of retrieving metadata documents and compiling them into the export document.
+	 * 
+	 * @author bbpennel
+	 * @date Jul 7, 2015
+	 */
+	public class XMLExportRunnable implements Runnable {
+		private final String user;
+		private final AccessGroupSet groups;
+		private final XMLExportRequest request;
+
+		public XMLExportRunnable(XMLExportRequest request, String user, AccessGroupSet groups) {
+			this.user = user;
+			this.groups = groups;
+			this.request = request;
+		}
+
+		@Override
+		public void run() {
+			long startTime = System.currentTimeMillis();
+			String seperator = System.getProperty("line.separator");
+			byte[] seperatorBytes = System.getProperty("line.separator").getBytes();
+
+			try {
+				GroupsThreadStore.storeGroups(groups);
+				GroupsThreadStore.storeUsername(user);
+
+				File mdExportFile = File.createTempFile("md_export", ".xml");
```

Comment:
It might be preferable to name this "xml_export" or something else similar to the controller class name.